nfsstat/check.py

```diff
@@ -0,0 +1,131 @@
+
+# stdlib
+import os
+
+# 3rd party
+
+# project
+from checks import AgentCheck
+from utils.subprocess_output import get_subprocess_output
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'nfsstat'
+
+class NfsStatCheck(AgentCheck):
+
+    metric_prefix = 'system.nfs.'
+
+    def __init__(self, name, init_config, agentConfig, instances):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        # if they set the path, use that
+        if init_config.get('nfsiostat_path'):
+            self.nfs_cmd = init_config.get('nfsiostat_path')
+        else:
+            # if not, check if it's installed in the opt dir, if so use that
+            if os.path.exists('/opt/datadog-agent/embedded/sbin/nfsiostat'):
+                self.nfs_cmd = '/opt/datadog-agent/embedded/sbin/nfsiostat'
+            # if not, then check if it is in the default place
+            elif os.path.exists('/usr/local/sbin/nfsiostat'):
+                self.nfs_cmd = '/usr/local/sbin/nfsiostat'
+            else:
+                raise Exception('nfsstat check requires nfsiostat be installed, please install it (through nfs-utils) or set the path to the installed version')
+
+    def check(self, instance):
+        stat_out, err, _ = get_subprocess_output(self.nfs_cmd, self.log)
+        all_devices = []
+        this_device = []
+        for l in stat_out.splitlines():
+            if not l:
+                continue
+            elif l.find('mounted on') >= 0 and len(this_device) > 0:
+                # if it's a new device, create the device and add it to the array
+                device = Device(this_device, self.log)
+                all_devices.append(device)
+                this_device = []
+            this_device.append(l.strip().split())
+
+        # Add the last device into the array
+        device = Device(this_device, self.log)
+        all_devices.append(device)
+
+        for device in all_devices:
+            device.send_metrics(self.gauge)
+
+class Device(object):
+
+    def __init__(self, device_data, log):
+        self.log = log
+
+        self._device_data = device_data
+        self._parse_device_header()
+        self._parse_tags()
+
+        self._parse_ops()
+        self._parse_read_data()
+        self._parse_write_data()
+
+    def _parse_device_header(self):
+        self._device_header = self._device_data[0]
+        self.log.info(self._device_header)
+        self.device_name = self._device_header[0]
+        self.mount = self._device_header[-1][:-1]
+        self.nfs_server = self.device_name.split(':')[0]
+        self.nfs_export = self.device_name.split(':')[1]
+
+    def _parse_ops(self):
+        ops = self._device_data[2]
+        self.ops = float(ops[0])
+        self.rpc_bklog = float(ops[1])
+
+    def _parse_read_data(self):
+        read_data = self._device_data[4]
+        self.read_ops = float(read_data[0])
+        self.read_kb = float(read_data[1])
+        self.read_kb_per_s = float(read_data[2])
```

Comment:
From the manual page: `This is the number of kB read per each operation` not per second